src/EventStore.Core/Services/Storage/ReaderIndex/EventFilter.cs

```diff
@@ -173,10 +173,18 @@ private sealed class EventTypeRegexStrategy : IEventFilter {
 				_isAllStream = isAllStream;
 				_expectedRegex = new Regex(expectedRegex, RegexOptions.Compiled);
 			}
-
-			public bool IsEventAllowed(EventRecord eventRecord) =>
-				(!_isAllStream || DefaultAllFilter.IsEventAllowed(eventRecord)) &&
-				_expectedRegex.Match(eventRecord.EventType).Success;
+			public bool IsEventAllowed(EventRecord eventRecord) {
+					if (_expectedRegex.ToString().Contains("^[^\\$].*")) {
+						return (!_isAllStream || DefaultAllFilter.IsEventAllowed(eventRecord)) &&
+						       _expectedRegex.Match(eventRecord.EventType).Success	&&
+						       (!eventRecord.EventStreamId.StartsWith("$persistentsubscription")) &&
```

Comment:
This logic should be in the `NonSystemStreamStrategy` rather than on the regex strategy

Reply:
Unfortunately, making changes to NonSystemStreamStrategy won't work as expected. The NonSystemStreamStrategy is not called when using the FilterSystemEvents() method on the client side. The filtering criteria for NonSystemStreamStrategy  is for filtering out all streams starting with '$', whereas the filtering criteria for FilterSystemEvents() is based upon event types (event types starting with '$') instead of stream names.